### PR TITLE
fix: Must not be able to start Job Card if it is related to Work Order that is not started yet

### LINKED ISSUE #29071: Cannot start Job Card if related to Work Order not started yet
Create a WO (from BOM with operation and Material transfer for WO)
Submit WO => Start WO => Stock transfer to WIP warehouse

Unfortunately you don't have enough stock to transfer to WIP, so you have WO in Submit status, Stock Entry in Draft, and Job card in draft.

On job card, you shouldn't be able to start counting time if work order is not started.

### PR DESCRIPTION
**Pull request by FHenry**
related to #29071

### CODE REVIEW

**Comment by FHenry** [comment]
Hi @marination Do you need more explanation from me about this PR ?

**Comment by johnlongland** [comment]
It will be great if this can be included into the core. Any idea when this might happen ?

File: erpnext/manufacturing/doctype/job_card/job_card.js [hunk 1]
```diff
@@ -73,7 +73,24 @@ frappe.ui.form.on('Job Card', {
 		if (frm.doc.docstatus == 0 && !frm.is_new() &&
 			(frm.doc.for_quantity > frm.doc.total_completed_qty || !frm.doc.for_quantity)
 			&& (frm.doc.items || !frm.doc.items.length || frm.doc.for_quantity == frm.doc.transferred_qty)) {
-			frm.trigger("prepare_timer_buttons");
+
+			//if Job Card is link to Work Order, the job card must not be able to start if Work Order not "Started"
+			// and if stock mvt for WIP is required
+			if (frm.doc.work_order) {
+				frappe.db.get_value('Work Order', frm.doc.work_order, 'skip_transfer').then((r) => {
```
Comment (marination) [inline erpnext/manufacturing/doctype/job_card/job_card.js]: Lets get whatever we need in one db call. `result` will have to be accessed as per whatever is returned (`r.message` wont be required)
```suggestion
				frappe.db.get_value('Work Order', frm.doc.work_order, ['skip_transfer', 'status']).then((result) => {
```

File: erpnext/manufacturing/doctype/job_card/job_card.js [hunk 2]
```diff
@@ -73,7 +73,24 @@ frappe.ui.form.on('Job Card', {
...
+					if (r.message.skip_transfer !== 1 ) {
```
Comment (marination) [inline erpnext/manufacturing/doctype/job_card/job_card.js]: The last else can be eliminated by combining both if conditions.
Also does "Completed" work order need a timer button?
```suggestion
					if (result.skip_transfer !== 1  && result.status == 'In Process') {
					     frm.trigger("prepare_timer_buttons");
					}
```

**Comment by FHenry** [comment]
@marination I've follow your instruction, and it's more readable like that. But result is not enought, we need result.message, like is it done line 95. It seems that frappe.db.get_value return result as r.message

**Review by marination — DISMISSED** [review]

**Review by marination — APPROVED** [review]

**Comment by FHenry** [comment]
@Mergifyio backport version-13-hotfix

**Pull request merged**